addons/website_slides/controllers/main.py

```diff
@@ -390,6 +421,82 @@ def slide_get_image(self, slide_id, field='image_medium', width=0, height=0, cro
+
+    @http.route('/get/quiz', type="json", auth="public", website=True)
+    def get_quiz(self, **kw):
+        if 'slide_id' in kw:
+            slide = request.env['slide.slide'].browse(kw['slide_id'])
+            slide_partner = request.env['slide.slide.partner'].search([('slide_id', '=', slide.id), ('partner_id', '=', request.env.user.partner_id.id)])
+            possible_points = [slide.quiz_first_attempt_reward,slide.quiz_second_attempt_reward,slide.quiz_third_attempt_reward, slide.quiz_fourth_attempt_reward]
+            points = 0
+            if slide_partner.quiz_attempts_count < len(possible_points):
+                points = possible_points[slide_partner.quiz_attempts_count]
+            else:
+                points = possible_points[len(possible_points)-1]
+            res = {
+                'questions':[{'title': question.question,'id': question.id,'answers': [{'text': answer.text_value, 'correct':answer.is_correct,'id': answer.id} for answer in question.answer_ids]} for question in slide.question_ids ],
+                'nb_attempts': slide_partner.quiz_attempts_count if slide_partner else 0,
+                'possible_rewards': possible_points,
+                'reward': points
+            }
+            return res
+
+    @http.route('/slide/quiz/submit', type="json", auth="user", website=True)
+    def submit_quiz(self, slide_id, answer_ids,**kw):
+        slide = request.env['slide.slide'].browse(slide_id)
+        answers = request.env['slide.answer'].browse(answer_ids)
+        good_answers = answers.search([('id', 'in', answer_ids),('is_correct', '=', True)])
+        bad_answers = answers - good_answers
+        slide_partner = request.env['slide.slide.partner'].search([('slide_id', '=', slide_id),('partner_id', '=', request.env.user.partner_id.id)])
+        possible_points = [slide.quiz_first_attempt_reward,slide.quiz_second_attempt_reward,slide.quiz_third_attempt_reward, slide.quiz_fourth_attempt_reward]
+        points = 0
+        if not slide_partner:
+            slide_partner = request.env['slide.slide.partner'].sudo().create({
+                "slide_id": slide.id,
+                "partner_id": request.env.user.partner_id.id,
+                "channel_id": slide.channel_id.id,
+                "completed": False
+            })
+        if not slide_partner.completed:
+            if slide_partner.quiz_attempts_count < len(possible_points):
+                points = possible_points[slide_partner.quiz_attempts_count]
+            else:
+                points = possible_points[len(possible_points)-1]
+            slide_partner.sudo().write({
+                'quiz_attempts_count': slide_partner.quiz_attempts_count if not bad_answers else slide_partner.quiz_attempts_count + 1,
+                'points_won': points if not bad_answers else 0,
+                'completed':  not bad_answers
+            })
+            return {
+                'goodAnswers':[good_answer.id for good_answer in good_answers],
```

Comment:
good_answers.ids, same for bad_answers.
Also underscore notation as I think jem already suggested.